hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -1393,6 +1417,47 @@ def group_by(group, agg_expr) -> DictExpression:
     return _agg_func.group_by(group, agg_expr)
 
 
+@typecheck(f=func_spec(1, expr_any),
+           array=expr_array())
+def array_agg(f, array):
+    """Aggregate an array element-wise using a user-specified aggregation function.
+
+    Examples
+    --------
+    Start with a range table with an array of random boolean values:
+
+    >>> ht = hl.utils.range_table(100)
+    >>> ht = ht.annotate(arr = hl.range(0, 5).map(lambda _: hl.rand_bool(0.5)))
+
+    Aggregate to compute the fraction ``True`` per element:
+
+    >>> ht.aggregate(hl.agg.array_agg(lambda element: hl.agg.fraction(element), ht.arr))  # doctest: +NOTEST
```

Comment:
you can test this if you'd like by setting a seed in the call to `hl.rand_bool`.

Reply:
I don't think we guarantee that the same seed will produce the same output across different Hail versions, though. Certainly within the same version, yes.